src/main/java/com/fzi/externalcontrol/impl/ExternalControlInstallationNodeContribution.java

```diff
@@ -65,8 +66,13 @@ public boolean isDefined() {
 
   @Override
   public void generateScript(ScriptWriter writer) {
-    // Make sure the program gets requested at least once
-    urScriptProgram = "";
+    urScriptHeader = "";
+    urScriptControlLoop = "";
+    RequestProgram sender = new RequestProgram(getHostIP(), getCustomPort());
+    String urScriptProgram = sender.sendCommand("request_program\n");
+
+    findHeaderAndControlLoop(urScriptProgram);
+    writer.appendRaw(urScriptHeader);
```

Comment:
With that change, I think we might end up with a non-working script at some situations:

I haven't understood completely in which situations the script code is generated, but I have the feeling that this doesn't necessarily happen, when the program gets started. If the program was started without a connected driver, the string `urScriptProgram` will contain an error message from [here](https://github.com/UniversalRobots/Universal_Robots_ExternalControl_URCap/blob/master/src/main/java/com/fzi/externalcontrol/impl/RequestProgram.java#L86). 

Thus, `findHeaderAndControlLoop` will return the complete string which is not problematic, but I think this change makes the program flow even more confusing.

Even worse, having this fetched inside the `generateScript` function of the InstallationNodeContribution makes this code run in each program, once this URCap is installed. Which is why I moved it in [this commit](https://github.com/UniversalRobots/Universal_Robots_ExternalControl_URCap/commit/c6587375092e1cee9af3323e7bdfcd6b7ad8d64c).

Reply:
I agree that we should not request the program every time once the program is installed, I was actually not aware of that. I will change it so that program only gets requested once the urcap is actually part of the program.   

The script code is generated every time the program is started and it gets executed when it is reached in the program execution. If we cannot connect to the driver, the string is parsed to the program node meaning that the popup will show, once the urcap program node is reached in the program execution. Therefore depending on the program structure the popup can be showed right away, but if the program consist of multiple nodes before the urcap, those nodes will be executed before the popup message.

Reply:
Isn't the script code also being generated once the program gets saved?

Otherwise we could create the popup during script generation from the URCap.

Reply:
Correct, it is also being generated when the program is saved

Reply:
Raising a popup there would probably be confusing, as you would have to have a running `ScriptSender` instance when creating the program on the TP. So, we'll leave it like this.